py/miniconf-mqtt/miniconf/miniconf.py

```diff
@@ -98,3 +139,46 @@ async def command(self, path, value, retain=False, timeout=5):
         result = await asyncio.wait_for(fut, timeout)
         if result['code'] != 0:
             raise MiniconfException(result['msg'])
+
+
+    def list_paths(self):
+        """ Get a list of all the paths available on the device. """
+        self._topics = []
+        fut = asyncio.get_running_loop().create_future()
+
+        request_id = uuid.uuid1().hex.encode()
+        assert request_id not in self.inflight
+        self.inflight[request_id] = ([], fut)
+
+        self.client.publish(f'{self.prefix}/list', payload='',
+                            correlation_data=request_id,
+                            response_topic=self.response_topic)
+        return fut
+
+
+    async def get(self, path, timeout=5):
```

Comment:
Do we want to similarly remove the timeout for `set` then? I'm confused why we wouldn't want a timeout in these.

Reply:
Yes. Timeouts should be higher level things.
Because the user can and should wrap this code in a timeouter themselves. https://github.com/quartiq/miniconf/pull/124#discussion_r1044348759